packages/lib-sourcify/src/lib/types.ts

```diff
@@ -159,10 +159,58 @@ export interface RecompilationResult {
   immutableReferences: ImmutableReferences;
 }
 
+export type Transformation = {
+  type: 'insert' | 'replace';
+  reason: 'constructor' | 'library' | 'immutable' | 'auxdata';
+  offset: number;
+  id?: string;
```

Comment:
What's the id for?

Reply:
valid for the case of auxdata transformation, it is the index of the auxdata (to handle multiple auxdata), in theory we could use the name of the contract instead of an number.